src/tile.cpp

```diff
@@ -742,27 +744,28 @@ ReturnValue Tile::queryRemove(const Thing& thing, uint32_t count, uint32_t flags
 	return RETURNVALUE_NOERROR;
 }
 
-Tile* Tile::queryDestination(int32_t&, const Thing&, Item** destItem, uint32_t& flags)
+std::shared_ptr<Thing> Tile::queryDestination(int32_t&, std::shared_ptr<const Thing>, std::shared_ptr<Item>& destItem,
+                                              uint32_t& flags)
 {
-	Tile* destTile = nullptr;
-	*destItem = nullptr;
+	std::shared_ptr<Tile> destTile = nullptr;
```

Comment:
Could this be `const auto`
```suggestion
	auto destTile = std::shared_ptr<Tile>(nullptr);
```